dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPIImpl.java

```diff
@@ -462,4 +465,32 @@ public boolean canDefaultPageToDefaultLanguage () {
     public boolean canDefaultFileToDefaultLanguage() {
         return Config.getBooleanProperty("DEFAULT_FILE_TO_DEFAULT_LANGUAGE",true);
     }
+
+	/**
+	 * {@inheritDoc}
+	 * @return
+	 */
+    @Override
+	public Language makeDefault(Long languageId, final User user)
+			throws DotDataException, DotSecurityException {
+    	if(!user.isAdmin()){
+    		throw new DotSecurityException("Only admin users are allowed to perform a default language switch.");
+		}
+		factory.makeDefault(languageId);
```

Comment:
you may need to double check that the languageId already exists before save it, otherwise throw an exception

Reply:
I think that's exactly the intent. There's a constraint that guarantees that the language inserted has to exist in the Language table.  If I send an invalid value I should get an exception